fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -623,20 +633,23 @@ void FixedLagSmoother::setDiagnostics(diagnostic_updater::DiagnosticStatusWrappe
 
     // Add time since the last optimization request time. This is useful to detect if no transactions are received for
     // too long
+    bool got_deadline_from_mutex = false;
     auto optimization_deadline = decltype(optimization_deadline_)();
     {
       const std::unique_lock<std::mutex> lock(optimization_requested_mutex_, std::try_to_lock);
       if (lock)
       {
         optimization_deadline = optimization_deadline_;
+        got_deadline_from_mutex = true;
       }
     }
 
-    if (!optimization_deadline.isZero())  // This is zero for the default-constructed optimization_deadline object
+    if (got_deadline_from_mutex)
```

Comment:
Seems like another spot where `rclcpp::Time::is_valid()` or mabye `rclcpp::Time::is_zero()` would be valuable.

Reply:
https://github.com/locusrobotics/fuse/pull/283/commits/9a1142173a60e2d50cb7bb40180b0ff85c8f486a